app/services/voting/state.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module Voting
+  class State
+    attr_reader :event
+
+    def initialize(control_strategy:, event:)
+      @control_strategy = control_strategy
+      @event = event
+    end
+
+    delegate :username,
+             :user_started_voting?,
+             :user_finished_voting?,
+             :current_award_id,
+             to: :@control_strategy
+
+    def ordered_projects
+      @_ordered_projects ||= projects.order("ideas.name")
```

Comment:
With the delegation of `name` to `idea` would the following work ?
```suggestion
      @_ordered_projects ||= projects.order("name")
```

Reply:
Apparently not. I just tried it and got:

```
Error:
       PG::UndefinedColumn: ERROR:  column "name" does not exist
       LINE 1: ...OM "projects" WHERE "projects"."event_id" = $1 ORDER BY name                    
```

I guess the delegation only works in Ruby and isn't propagated by `ActiveRecord` into SQL.

Reply:
Oh good to know. Thank you for testing!